### DISCUSSION TITLE
Guidelines for LIDVIDs for regression tests

### DISCUSSION THREAD

**Discussion by wkiri** [pds-data-dictionaries/PDS-Data-Dictionaries.github.io#14] [Ideas]
Inspired by our review of the namespaces, I suggest that we identify additions to the guidelines for creating test cases that are provided here:
https://pds-data-dictionaries.github.io/development/ldd-test.html

1. I recommend including guidelines for specifying LIDVIDs (`logical_identifier`) for test cases (test `.xml` labels).  The current examples in the `ldd-template` repo use `urn:nasa:pds:bundle:collection:product`, but it seems useful to encourage stewards to use a different LIDVID for each test case, ideally something meaningful.  One idea would be

`urn:nasa:pds:namespace_<namespace>:test:test_invalid_value_FAIL`

for a test label named `test_invalid_valid_FAIL.xml`.  Other thoughts/suggestions?

2. I recommend including other instructions such as:
  * Make test `.xml` filenames as specific as possible
  * Include a comment at the top of the `.xml` file indicating the purpose of the test
  * Include a comment near the line(s) inside the file that cause failure (for tests expected to fail)
  * Start with a minimal working label and only introduce content needed to exercise the desired test, to make the test cases easier to read

3. If this discussion results in changes to the guidelines for test cases, we'll want to make sure that the test cases in the `ldd-template` repo follow the guidelines.  
Relatedly, one of the example test cases provided in the `ldd-template` repo does not follow the current guidelines (to specify [IM_VERSION] instead of a specific information model version):
https://github.com/pds-data-dictionaries/ldd-template/blob/main/test/test1_FAIL.xml
(unless this was deliberate in order to make it fail for other IM versions?).  I can't seem to create an issue for the `ldd-template` repository, so I'm noting it here for @jordanpadams :) .

Have you identified other good practices for test cases that we can/should share and adopt?  Please add your comments here.

**Comment by jstone-psi** [comment]
I've incorporated some of your suggestions into the survey dictionary at (https://github.com/pds-data-dictionaries/ldd-survey/tree/tests/standardize-test-names). I think that you can do a little bit more with the collection names, as well. In this branch, I'm trying `data_test_<test_suite>` for the collection id.

Additionally, I've added a README.md file to the test directory that specifies what each test does, so you can learn about all of the tests in a single centralized location. Putting a comment near the expected failure would still be a good idea. And so would putting a comment at the top of the file.

**Reply by wkiri** [reply]
Excellent idea to add a test suite `README.md`, @jstone-psi !

I also like your `data_test_<test_suite>` convention (or just `test_<test_suite>`?).

Btw, it looks like this test is missing an underscore in the collection name, `datatest_rollover_fwhm_units`:

https://github.com/pds-data-dictionaries/ldd-survey/blob/1d4ae5c2c578a32ddf40a3feff574bc8fb8f336b/test/test_rollover_fwhm_units/test_rollover_fwhm_units_FAIL.xml#L16

**Comment by jstone-psi** [comment]
Wow, I was sure that the collection type had to be in the collection_id, but I can't find anything that supports that. In that case, `test_<test_suite>` should be fine.
